File: src/main/scala/com/github/julekarenalender/log/Logger.scala
package com.github.julekarenalender.log

object Logger extends Logger with ConsoleLogger

trait Logger {
  def enableDebug()
  def debug(logLine: String)
  def info(logLine: String)
  def error(logLine: String)
  def error(logLine: String, thrown: Throwable)
}

File: src/main/kotlin/com/github/julekarenalender/logger/Logger.kt
package com.github.julekarenalender.logger

class Logger(val isDebug: Boolean = false) {

    fun debug(str: String) {
        if (isDebug) {
            println("DEBUG: $str")
        }
    }

    fun info(str: String) {
        println(str)
    }

    fun warn(str: String) {
        println("Ups! $str")
    }

    fun error(str: String) {
        println("Oh no! $str")
    }

    fun error(str: String, thr: Throwable) {
        println("Oh no! $str ${thr.javaClass}: ${thr.message}")
    }

}